File: tests/Generator/resource/csharp/complex/MapDefinitionType.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

/// <summary>
/// Represents a map which contains a dynamic set of key value entries of the same type
/// </summary>
public class MapDefinitionType : CollectionDefinitionType
{
    [JsonPropertyName("type")]
    public new string? Type { get; set; } = "map";

}

File: tests/Generator/resource/php/complex/MapDefinitionType.php
<?php

declare(strict_types = 1);

namespace TypeAPI\Model;

use PSX\Schema\Attribute\Description;

#[Description('Represents a map which contains a dynamic set of key value entries of the same type')]
class MapDefinitionType extends CollectionDefinitionType implements \JsonSerializable, \PSX\Record\RecordableInterface
{
}

